mcs/class/corlib/System.Runtime.Remoting.Messaging/MonoMethodMessage.cs

```diff
@@ -109,19 +109,30 @@ public MonoMethodMessage (MethodBase method, object [] out_args)
 				args = null;
 		}
 
-		public MonoMethodMessage (Type type, string method_name, object [] in_args)
+		internal MonoMethodMessage (MethodInfo minfo, object [] in_args, object [] out_args)
 		{
-			// fixme: consider arg types
-			MethodInfo minfo = type.GetMethod (method_name);
-			
-			InitMessage ((MonoMethod)minfo, null);
+			InitMessage ((MonoMethod)minfo, out_args);
 
 			int len = in_args.Length;
 			for (int i = 0; i < len; i++) {
 				args [i] = in_args [i];
 			}
 		}
+
+		private static MethodInfo getMethodInfo (Type type, string method_name)
```

Comment:
Meaning it should be: static MethodInfo GetMethodInfo (Type type, string methodName)